src/xtgeo/grid3d/_egrid.py

```diff
@@ -601,23 +612,25 @@ def coord(self) -> np.ndarray:
         return self.global_grid.coord
 
     @coord.setter
-    def coord(self, value: np.ndarray):
+    def coord(self, value: np.ndarray) -> None:
         self.global_grid.coord = value
 
     @property
     def zcorn(self) -> np.ndarray:
         return self.global_grid.zcorn
 
     @zcorn.setter
-    def zcorn(self, value: np.ndarray):
+    def zcorn(self, value: np.ndarray) -> None:
         self.global_grid.zcorn = value
 
     @property
     def actnum(self) -> np.ndarray | None:
         return self.global_grid.actnum
 
     @classmethod
-    def from_file(cls, filelike, fileformat: str = None):
+    def from_file(
+        cls, filelike: FileLike, fileformat: str | None = None
+    ) -> EGridReader:
```

Comment:
It's a classmethod so it should return an `EGrid` (the return value sneaks a `read()` at the end)

Reply:
wow nice catch, strange that mypy did not see this.